gui_scripts/gui_helpers/general_helpers.py

```diff
@@ -202,3 +202,178 @@ def stop(self):
             self.paused = False
             self.output_hints_signal.emit('Stopping simulation from thread')
         self.pause_condition.wakeOne()
+
+
+class DirectoryTreeView(QtWidgets.QTreeView):
+    """
+    Sets up a new directory tree view
+    """
+    item_double_clicked_sig = QtCore.pyqtSignal(QtCore.QModelIndex)
+
+    def __init__(self, file_model, parent=None):
+        super().__init__(parent)
+        self.setSelectionBehavior(QtWidgets.QTreeView.SelectRows)
+        self.setSelectionMode(QtWidgets.QTreeView.SingleSelection)
+
+        self.model = file_model
+        self.setModel(self.model)
+
+        self.copied_path = None
+        self.is_directory = False
+        self.is_cut_operation = False
+
+    def copy_item(self, source_index, cut=False):
```

Comment:
In our coding guidelines document, we must put variable types in our function or method arguments. For example, is source index an integer?